src/lib/Microsoft.Health.Fhir.Ingest.Template/HandlerProxyTemplateFactory.cs

```diff
@@ -12,8 +14,29 @@ public abstract class HandlerProxyTemplateFactory<TInput, TOutput> :
         IResponsibilityHandler<TInput, TOutput>
         where TOutput : class
     {
+        private readonly IList<string> _serializationErrors = new List<string>();
+
+        protected IList<string> SerializationErrors
+        {
+            get { return _serializationErrors; }
+        }
+
         public abstract TOutput Create(TInput jsonTemplate);
 
         TOutput IResponsibilityHandler<TInput, TOutput>.Evaluate(TInput request) => Create(request);
+
+        protected JsonSerializer GetJsonSerializer()
+        {
+            _serializationErrors.Clear();

```

Comment:
Coupling the clearing of the errors with getting the serialization doesn't seem correct to me.  It would be better to have errors part of the TOutput return object so error life cycle naturally

Reply:
Updated with the extension approach suggested by Dustin. Much clean and better now.